grails-app/services/org/pih/warehouse/core/LocalizationService.groovy

```diff
@@ -61,7 +58,7 @@ class LocalizationService {
      */
     Locale getCurrentLocale() {
         // fetch the locale of the current user; if there isn't one, use the default locale
-        return (RequestContextHolder.currentRequestAttributes().getSession().user?.locale ?:
+        return (WebUtils.retrieveGrailsWebRequest().getSession()?.user?.locale ?:
```

Comment:
Looking at this for the first time in a long time, I just realized that this probably could have been done using the AuthService.currentUser thread local. But we can leave this for later. Note that this throws an illegalStateException when accessed outside of a request, so we might want to test that. Another option would be to use RequestContextHolder?.requestAttributes?.session.user?.locale.

http://docs.grails.org/3.1.6/api/org/grails/web/util/WebUtils.html#retrieveGrailsWebRequest()

Reply:
@jmiranda I changed it